src/metrics.cc

```diff
@@ -38,7 +38,7 @@ int64_t HighResTimer() {
   timeval tv;
   if (gettimeofday(&tv, NULL) < 0)
     Fatal("gettimeofday: %s", strerror(errno));
-  return (int64_t)tv.tv_sec * 1000*1000 + tv.tv_usec;
+  return static_cast<int64_t>(tv.tv_sec) * 1000 * 1000 + tv.tv_usec;
```

Comment:
The type of 1000 should be explicitly set with ULL to avoid weird compiler bullshit.